src/idiomatic/leveraging-the-type-system/token-types/branded-02-phantomdata.md

```diff
@@ -0,0 +1,124 @@
+---
+minutes: 20
+---
+
+# `PhantomData` and Lifetime Subtyping (Branding 2/4)
+
+We need a mechanism to make lifetimes distinct enough from each other to not
+compile when we try to compare them.
+
+<!-- dprint-ignore-start -->
+```rust,editable
+use std::marker::PhantomData;
+
+#[derive(Default)]
+struct InvariantLifetime<'id>(PhantomData<&'id ()>); // The main focus
+
+struct Wrapper<'a> { value: u8, invariant: InvariantLifetime<'a> }
+
+fn lifetime_separator<T>(value: u8, f: impl for<'a> FnOnce(Wrapper<'a>) -> T) -> T {
+    f(Wrapper { value, invariant: InvariantLifetime::default() })
+}
+
+fn compare_lifetimes<'a>(left: Wrapper<'a>, right: Wrapper<'a>) {}
```

Comment:
Would "try_coerce_lifetimes", "are_coercible_lifetimes", or "check_common_type" be better names?

I would like to avoid using the word "compare" because that's just not a thing that the type system in Rust does.